src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -197,7 +197,7 @@ private async Task ProcessQueueAsync()
                     {
                         // Mutating requests block other requests from starting to ensure an up to date snapshot is used.
                         // Since we're explicitly awaiting exceptions to mutating requests will bubble up here.
-                        await work.StartRequestAsync(context, cancellationToken).ConfigureAwait(false);
+                        await WrapStartRequestTaskAsync(work.StartRequestAsync(context, cancellationToken)).ConfigureAwait(false);
```

Comment:
As per @davidwengier's suggestion I wrapped the mutating request too. @dibarbet let me know if there's some reason we shouldn't do that.

Reply:
@ryanbrandenburg the only concern I have is check that mutating request exceptions still bubble up to the queue (and don't get entirely caught by report non-fatal).

Reply:
I was thinking a flag could be used, to tell the implementor whether to rethrow, but perhaps it doesn't matter and they should just always rethrow? Unfortunately the only way to communicate that is with docs